packages/uploadthing/src/internal/validate-request-input.ts

```diff
@@ -0,0 +1,246 @@
+      );
+    }
+
+    if (!slug) {
+      logger.error("No slug provided in params:", params);
+      return yield* $(
+        new UploadThingError({
+          code: "BAD_REQUEST",
+          message: "No slug provided in params",
+        }),
+      );
+    }
+
+    if (slug && typeof slug !== "string") {
+      const msg = `Expected slug to be of type 'string', got '${typeof slug}'`;
+      logger.error(msg);
+      return yield* $(
+        new UploadThingError({
+          code: "BAD_REQUEST",
+          message: "`slug` must be a string",
+          cause: msg,
+        }),
+      );
+    }
+    if (actionType && typeof actionType !== "string") {
+      const msg = `Expected actionType to be of type 'string', got '${typeof actionType}'`;
+      logger.error(msg);
+      return yield* $(
+        new UploadThingError({
+          code: "BAD_REQUEST",
+          message: "`actionType` must be a string",
+          cause: msg,
+        }),
+      );
+    }
+    if (uploadthingHook && typeof uploadthingHook !== "string") {
+      const msg = `Expected uploadthingHook to be of type 'string', got '${typeof uploadthingHook}'`;
+      return yield* $(
+        new UploadThingError({
+          code: "BAD_REQUEST",
+          message: "`uploadthingHook` must be a string",
+          cause: msg,
+        }),
+      );
+    }
+
+    const apiKey = getApiKey(opts.opts.config?.uploadthingSecret);
+    if (!apiKey) {
+      const msg = `No secret provided, please set UPLOADTHING_SECRET in your env file or in the config`;
+      logger.error(msg);
+      return yield* $(
+        new UploadThingError({
+          code: "MISSING_ENV",
+          message: `No secret provided`,
+          cause: msg,
+        }),
+      );
+    }
+
+    if (!apiKey.startsWith("sk_")) {
+      const msg = `Invalid secret provided, UPLOADTHING_SECRET must start with 'sk_'`;
+      logger.error(msg);
+      return yield* $(
+        new UploadThingError({
+          code: "MISSING_ENV",
+          message: "Invalid API key. API keys must start with 'sk_'.",
+          cause: msg,
+        }),
+      );
+    }
+
+    if (utFrontendPackage && typeof utFrontendPackage !== "string") {
+      const msg = `Expected x-uploadthing-package to be of type 'string', got '${typeof utFrontendPackage}'`;
+      logger.error(msg);
+      return yield* $(
+        new UploadThingError({
+          code: "BAD_REQUEST",
+          message:
+            "`x-uploadthing-package` must be a string. eg. '@uploadthing/react'",
+          cause: msg,
+        }),
+      );
+    }
+
+    const uploadable = opts.opts.router[slug];
+    if (!uploadable) {
+      const msg = `No file route found for slug ${slug}`;
+      logger.error(msg);
+      return yield* $(
+        new UploadThingError({
+          code: "NOT_FOUND",
+          message: msg,
+        }),
+      );
+    }
+
+    if (
+      uploadthingHook !== "callback" && // TODO: Maybe just send callback as actionType? Or actionType as header?
```

Comment:
yeah... not sure why we did it this way. will need to maintain it for back compat now though :/

Reply:
prob not worth it - has worked fine until now + it's a clear separation of where it's coming from which i guess could be nice sometimes